### PR TITLE
propagate title and href by default

### LINKED ISSUE #617: The bin and group transform could propagate the title channel, if any, by default
E.g.,

```js
Plot.rectY(data, Plot.binX({y: "sum"}, {x: "date", y: "count", fill: "name", title: "name"}))
```

would be equivalent to

```js
Plot.rectY(data, Plot.binX({y: "sum", title: "first"}, {x: "date", y: "count", fill: "name", title: "name"}))
```

And possibly it could even be smart by showing the distinct values rather than just the first? Or it could be like `Fred (42)` and show both the distinct values and their respective counts by default? I feel like that would be very useful.

### PR DESCRIPTION
**Pull request by mbostock**
Fixes #617.

### CODE REVIEW

File: src/transforms/group.js [hunk 1]
```diff
@@ -266,6 +270,16 @@ const reduceFirst = {
...
 };
 
+const reduceTitle = {
+  reduce(I, X) {
+    return sort(rollup(I, V => V.length, i => X[i]), second)
+      .reverse()
+      .slice(0, 5)
+      .map(([key, value]) => `${key} (${value.toLocaleString("en-US")})`)
+      .join("\n");
+  }
```
Comment (Fil) [inline src/transforms/group.js:280]:
```suggestion
    const labels = sort(rollup(I, V => V.length, i => X[i]), second)
      .reverse();
    return labels
      .slice(0, 5)
      .map(([key, value], i) => i === 4 && labels.length > 5 ? "…" : `${key} (${value.toLocaleString("en-US")})`)
      .join("\n");
  }
```

to replace the 5th element by continuation dots if there are more?

Comment (mbostock) [reply src/transforms/group.js:280]: Probably worth counting the number of “other” then.

Comment (mbostock) [reply src/transforms/group.js]: If there are more than five distinct values, these are now combined into an “Other” category.

File: src/transforms/group.js [hunk 2]
```diff
@@ -266,6 +270,15 @@ const reduceFirst = {
   }
 };
 
+const reduceTitle = {
+  reduce(I, X) {
+    const groups = sort(rollup(I, V => V.length, i => X[i]), second);
+    const top = groups.slice(-5).reverse();
+    if (top.length < groups.length) top.push(["Other", sum(groups.slice(0, -5), second)]);
```
Comment (Fil) [inline src/transforms/group.js:277]:
```suggestion
    const N = 5;
    const groups = sort(rollup(I, V => V.length, i => X[i]), second);
    const top = groups.slice(-5).reverse();
    if (top.length < groups.length) top.push([groups.length === N + 1 ? groups[N][0] : "Other", sum(groups.slice(0, -N), second)]);
```

Comment (Fil) [reply src/transforms/group.js:277]: (if there are exactly 6 groups we can use the 6th's name.)

**Comment by Fil** [comment]
The locale (https://github.com/observablehq/plot/issues/384) should also apply to "Other".

File: src/transforms/group.js (hunk 2 above)
Comment (mbostock) [reply src/transforms/group.js]: Changed to show the number of other categories, as well as the number of constituent elements.

**Review by Fil — APPROVED** [review]
👍

**Pull request merged**